sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -578,17 +588,44 @@ def __init__(self, value_coder, timestamp_coder, window_coder):
 
   def encode_to_stream(self, value, out, nested):
     wv = value  # type cast
-    self._value_coder.encode_to_stream(wv.value, out, True)
     # Avoid creation of Timestamp object.
-    out.write_bigendian_int64(wv.timestamp_micros)
+    restore_sign = -1 if wv.timestamp_micros < 0 else 1
```

Comment:
Can you add a comment that this big hack is to maintain compatibility with the Java SDK and point to a jira issue to resolve this in the future?